ComponentKit/Core/Scope/CKComponentScopeHandle.mm

```diff
@@ -21,8 +22,13 @@ @implementation CKComponentScopeHandle
 {
   id<CKComponentStateListener> __weak _listener;
   Class _componentClass;
+  CKComponentController *_controller;
   CKComponentScopeRootIdentifier _rootIdentifier;
   BOOL _acquired;
+  // Temporarily stored reference to the specific component that acquired this handle. This forms a reference cycle
+  // that is broken in `resolve`.
+  CKComponent *_acquiredComponent;
```

Comment:
I think it's worthwhile calling out that the component `_acquiredComponent` references is likely not completely initialized. I'd hate to see a change later on that tries to pass this component along elsewhere before the handle has been fully resolved.